server/src/com/thoughtworks/go/config/ConfigConverter.java

```diff
@@ -134,7 +134,10 @@ public PluggableTask toPluggableTask(CRPluggableTask pluggableTask) {
 
     private void setCommonTaskMembers(AbstractTask task, CRTask crTask) {
         CRTask crTaskOnCancel = crTask.getOnCancel();
-        task.setCancelTask(crTaskOnCancel != null ? toAbstractTask(crTaskOnCancel) : null);
+        if(crTaskOnCancel != null)
```

Comment:
Good point - `AbstractTask.onCancelConfig` had already the kill-all task in there by default. 
I updated the commit, so now task is set only when `crTask.getOnCancel()` is null. Otherwise stays default.
@jyotisingh  please merge if you are fine with it now.